### ISSUE TITLE
Expecting Object, got: none while handling: UpdateNode

### ISSUE THREAD

**Issue by gurisko** [arangodb/arangodb#8197]
## My Environment

* __ArangoDB Version__:        3.4.2-1
* __Storage Engine__:             MMFiles
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   ArangoDB Starter
* __Configuration__:               -
* __Infrastructure__:               own
* __Operating System__:        Debian 8
* __Total RAM in your machine__:        8GB
* __Disks in use__:        HDD
* __Used Package__:              Debian or Ubuntu .deb

## Component, Query & Data

__Affected feature__: AQL

__AQL query__:
```
FOR reg IN regions
    COLLECT id = reg._id INTO g
    UPDATE g[0].reg WITH {region: true} IN regions
```

__Dataset__:
[Regions](https://github.com/arangodb/example-datasets/tree/master/Regions)

__Problem__:
```
Query: AQL: expecting 'Object', got: none while handling: UpdateNode (while executing)
```
This query worked fine in 3.3.19 but it doesn't work in 3.4.2-1.
It seems that the issue is somewhere with using `INTO g` because the following workaround works:
```
FOR reg IN regions
    COLLECT id = reg._id
    LET reg = FIRST(FOR r IN regions FILTER r._id == id RETURN r)
    UPDATE reg WITH {region: true} IN regions
```

__Expected result__:
The query works.

**Comment by jsteemann** [comment]
I just tried this in current 3.4 (3.4.3) and couldn't reproduce it.
I then tried 3.4.2 and was able to reproduce the issue.

So I think this problem has been fixed in 3.4.3 already. However, 3.4.3 is currently in QA stage and hasn't been released yet. But I expect it to be released within the next few days.

**Comment by sleto-it** [comment]
Hi,

Thanks for reporting this. Version 3.4.3 has been released

I am going ahead and closing this ticket now. Will you please add a new comment if further help is needed on this topic, so we can reopen the ticket?

Many Thanks,
